### PR TITLE
test(components): Unit test for Sagemaker Hosting component and new runtime

### PR DESCRIPTION
**Pull request by ananth102**
**Description of your changes:**

Introduction of upgrade and Endpoint related unit tests

components/aws/sagemaker/tests/unit_tests/tests/commonv2/*
components/aws/sagemaker/tests/unit_tests/tests/Model/*
components/aws/sagemaker/tests/unit_tests/tests/Endpoint/*
components/aws/sagemaker/tests/unit_tests/tests/EndpointConfig/*

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by rrrkharse — CHANGES_REQUESTED** [review]

File: components/aws/sagemaker/tests/unit_tests/tests/Endpoint/test_Endpoint_component.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+
+class EndpointComponentTestCase(unittest.TestCase):
+    REQUIRED_ARGS = [
+        "--region",
+        "us-west-1",
+        "--endpoint_config_name",
+        "test",
+        "--endpoint_name",
+        "test1",
```
Comment (ryansteakley) [inline components/aws/sagemaker/tests/unit_tests/tests/Endpoint/test_Endpoint_component.py:17]: nit: Can we have a more descriptive name for the test than test1?

File: components/aws/sagemaker/tests/integration_tests/utils/ack_utils.py [hunk 2]
```diff
@@ -16,31 +16,71 @@ def _get_resource(k8s_client, job_name, kvars):
...
+        return False
+def is_endpoint_deleted(
+    k8s_client, endpoint_name
+):
+    try:
+        response = _get_resource(k8s_client, endpoint_name, "endpoints")
+        if response:
+            return False
+    except:
+        return True
```
[the remark this answers is not in the document]
Comment (ananth102) [reply components/aws/sagemaker/tests/integration_tests/utils/ack_utils.py:86]: Need to handle the None case(404 is returned when the resource is not found though), there is a retry in test_endpoint

File: components/aws/sagemaker/Endpoint/src/Endpoint_spec.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    sagemaker_resource_name: Output
+
+
+class SageMakerEndpointSpec(
+    SageMakerComponentSpec[SageMakerEndpointInputs, SageMakerEndpointOutputs]
+):
+    INPUTS: SageMakerEndpointInputs = SageMakerEndpointInputs(
+        deployment_config=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            description="The deployment configuration for an endpoint, which contains the desired deployment strategy and rol",
```
[the remark this answers is not in the document]
Comment (ananth102) [reply components/aws/sagemaker/Endpoint/src/Endpoint_spec.py]: Trimmed all the descriptions down to the first sentence.

File: components/aws/sagemaker/tests/integration_tests/resources/config/ack-hosting-update/config.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  production_variants:
+    - variantName: variant-1
+      initialVariantWeight: 1.0
+      instanceType: ml.m5.xlarge
+      initialInstanceCount: 1
+  second_production_variants:
+    - variantName: variant-1
+      initialVariantWeight: 1.0
+      instanceType: ml.m5.large
+      initialInstanceCount: 1
```
[the remark this answers is not in the document]
Comment (ananth102) [reply components/aws/sagemaker/tests/integration_tests/resources/config/ack-hosting-update/config.yaml:26]: The goal of this test is to check if update in KFP is functional. KFP relies on the downstream dependency(ACK) for coverage.

File: components/aws/sagemaker/tests/unit_tests/tests/Endpoint/test_Endpoint_component.py (hunk 1 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/unit_tests/tests/Endpoint/test_Endpoint_component.py]: Done

**Review by ryansteakley — APPROVED** [review]

**Pull request closed without merging**